### PR TITLE
Various UI and lobby fixes

### PR DESCRIPTION
**Pull request by Vagabond**
See commits for details.

### CODE REVIEW

**Review by katajakasa — DISMISSED** [review]
lgtm.

File: src/game/gui/label.c [hunk 1]
```diff
@@ -45,9 +45,10 @@ component *label_create_with_width(const text_settings *tconf, const char *text,
 
     int tsize = text_char_width(tconf);
     int h = text_find_line_count(tconf, max_width / tsize, 0, strlen(text), text);
+    int width = text_find_max_strlen(tconf, max_width / tsize, text);
 
     // fonts are all 8 high?
-    component_set_size_hints(c, max_width, h * 8);
+    component_set_size_hints(c, width * tsize , h * 8);
```
Comment (Nopey) [inline src/game/gui/label.c:51]: `text_find_max_strlen` returns the length of the first line of wrapped text, not the maximum length of any line of text.

Perhaps text_find_line_count needs to be modified to also provide the caller with the longest line's length?

File: src/utils/str.c [hunk 2]
```diff
@@ -29,29 +29,31 @@ static void str_resize_buffer(str *dst, size_t size) {
...
-    size_t size_with_zero = size + 1;
-    if(size_with_zero > STR_STACK_SIZE) {
-        // New size is larger than the stack buffer; do malloc.
-        if(dst->data == NULL) {
-            // Old string is in stack, move to heap
-            dst->data = omf_malloc(size_with_zero);
-            memcpy(dst->data, dst->small, dst->len);
-            dst->data[dst->len] = 0;
-            dst->small[0] = 0;
+    if(size != dst->len) {
```
Comment (Nopey) [inline src/utils/str.c:32]:
```suggestion
    if(size == dst->len) {
        return;
    }
```

**Review by Nopey — CHANGES_REQUESTED** [review]
label_create_with_width change regresses kreissack toopathetic dialog

File: testing/test_text_render.c [hunk 3]
```diff
@@ -83,46 +86,64 @@ static void test_text_find_line_count_newsroom(void) {
...
                     9);
+    CU_ASSERT_EQUAL(6, longest);
+
     // after reaching max_lines, word wrap stops (and no more words are rendered, but we don't care about that behavior)
-    CU_ASSERT_EQUAL(
-        text_find_line_count_(&tconf, 2, 999, "one two three four five six seven eight nine ten eleven twelve"), 9);
+    CU_ASSERT_EQUAL(text_find_line_count_(&tconf, 2, 999,
+                                          "one two three four five six seven eight nine ten eleven twelve", &longest),
+                    9);
+    CU_ASSERT_EQUAL(2, longest);
```
Comment (Nopey) [inline testing/test_text_render.c:146]: The last line of this test's text will be more than 2 characters long

Comment (Nopey) [reply testing/test_text_render.c:146]: wait no im still waking up that's not real

**Comment by Nopey** [comment]
changes look like kreissack's dialog should still work, just gonna spin up the game rq

**Comment by Nopey** [comment]
image
yeah, kreissack's good

**Pull request merged**